librarian_server/api/instances.py

```diff
@@ -0,0 +1,105 @@
+"""
+API endpoints for instance management.
+
+Obviously, you need admin permissions to use these endpoints, except
+the self-password change.
+"""
+
+from fastapi import APIRouter, Depends, Response, status
+from sqlalchemy import desc, select
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Session
+
+from hera_librarian.authlevel import AuthLevel
+from hera_librarian.models.instances import (
+    InstanceAdministrationDeleteRequest,
+    InstanceAdministrationChangeResponse,
+)
+
+from ..database import yield_session
+from ..logger import log
+from ..orm.instance import RemoteInstance, Instance
+from ..settings import server_settings
+from .auth import AdminUserDependency, ReadonlyUserDependency, UnauthorizedError
+
+router = APIRouter(prefix="/api/v2/instances")
+
+
+@router.post(
```

Comment:
I would prefer it if these endpoints were in admin.py and were under admin/instance/delete_local and /delete_remote, rather than being in their own file.